clients/mobile/src/myActivity/context/MyActivityProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useState, useEffect, ReactNode } from 'react'
+import { useFetchClient } from '../../setup/clients/useFetchClient'
+import { fetchDonationList } from '../../donationWorkflow/donationService'
+import { DonationData, formatDonations, extractErrorMessage } from '../../donationWorkflow/donationHelpers'
+import { useUserProfile } from '../../userWorkflow/context/UserProfileContext'
+
+export interface MyActivityContextType {
+  donationPosts: DonationData[];
+  errorMessage: string | null;
+  loading: boolean;
+  fetchDonationPosts: () => Promise<void>;
+}
+
+const defaultContextValue = {
+  donationPosts: [],
+  errorMessage: null,
+  loading: false,
+  fetchDonationPosts: async() => { }
+}
+
+export const MyActivityContext = createContext<MyActivityContextType>(defaultContextValue)
+
+export const MyActivityProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const { userProfile } = useUserProfile()
+  const [donationPosts, setDonationPosts] = useState<DonationData[]>([])
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const fetchClient = useFetchClient()
+
+  useEffect(() => { void fetchDonationPosts() }, [userProfile])
+
+  const fetchDonationPosts = async(): Promise<void> => {
+    setLoading(true)
+    try {
+      const response = await fetchDonationList({}, fetchClient)
+      if (response.data !== undefined && response.data.length > 0) {
+        const formattedDonations = formatDonations(response.data, userProfile.name)
+        setDonationPosts(formattedDonations)
+      } else {
+        setDonationPosts([])
+      }
+    } catch (error) {
+      const errorMessage = extractErrorMessage(error)
+      setErrorMessage(errorMessage)
+    } finally {
+      setLoading(false)
```

Comment:
same, i feel like we can simplify this

Reply:
Create a hook `useFetchData.ts` for handling API calls so that we can eliminate all repeated code, place it in a central location, and make the code cleaner. [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/295/commits/f85fc2ea6106e588e1545dd6250cc3b1105a6e53)